content/blogs/gui-setup-part3-first-testing-problems.md

```diff
@@ -0,0 +1,275 @@
+
+
+## Finishing and testing the partitioning page UI
+
+The first week was devoted to finally putting together the code for manipulating partitions from the user interface, [using the new workflow](/blogs/gui-setup-part2-partitioning#workflow-change-in-the-gui-mode-setup), and testing it.
+At the UI level, this means:
+- verifying that inserting and deleting partitions correctly inserts or deletes their entries from the list;
+- that two consecutive empty space entries get merged together;
+- and that supplemental partition creation properties (like the selected formatting options) stay associated to the correct partition (and don't get freed, until the correct partition entry is deleted).
+
+
+## Putting everything together, testing...
+
+In this test-case scenario, running ReactOS GUI setup requires booting an _all-in-one_ CD ISO image.
+This is an image that contains both the installation files, that we usually distribute as part of the _bootcd_, and the rest of the OS to boot from a graphical environment, that we distribute as a _livecd_.
+
+{{< gallery >}}
+{{< figure link="/img/blogs/hbelusca_gui-setup/hybridcd_boot_menu.png" src="/img/blogs/hbelusca_gui-setup/hybridcd_boot_menu.png" caption="All-in-One ISO Boot menu (subject to change in the future)">}}
+{{< /gallery >}}
+
+Selecting one of the `ReactOS Live` (or `ReactOS Setup and Live`) entries, execution proceeds to the usual language-selection screen,
+followed by the screen where to choose between starting the usual LiveCD desktop environment, or the ReactOS GUI installer.
+
+{{< gallery >}}
+{{< figure link="/img/blogs/hbelusca_gui-setup/hybridcd_lang_select.png" src="/img/blogs/hbelusca_gui-setup/hybridcd_lang_select.png" caption="General language and keyboard layout selection">}}
+{{< figure link="/img/blogs/hbelusca_gui-setup/hybridcd_run_install_select.png" src="/img/blogs/hbelusca_gui-setup/hybridcd_run_install_select.png" caption="Want to run the LiveCD environment, or start the GUI setup?">}}
+{{< /gallery >}}
+
+Note that the latter can also be started from the LiveCD desktop.
+
+{{< gallery >}}
+{{< figure link="/img/blogs/hbelusca_gui-setup/hybridcd_guisetup_from_desktop.png" src="/img/blogs/hbelusca_gui-setup/hybridcd_guisetup_from_desktop.png" caption="The GUI setup can also be run from the LiveCD desktop">}}
+{{< /gallery >}}
+
+The language-selection dialog, shown above, currently also affects the language and keyboard layout that will be used for the new ReactOS installation (_NOTE: This is subject to change in the future._)
+Installation then starts with the usual welcome screen, followed by the choice of supported hardware presets (computer, display and keyboard types).
+Then, follows the list of disks and their partitions detected on the system.
+Currently, partitions can be only created on already-initialized MBR disks -- support for GPT disks and not-yet-initialized disks is planned to be added in the future.
```

Comment:
"are planned" --> here I meant the support (in singular), so I think, "is planned" would be more correct?

Reply:
> Additional support for GPT disks and uninitialized disks is planned in the future.

Planned *for* the future would sound better, but otherwise sounds good to me :) I was thinking of support for GPT disks and support for uninitialized disks as separate features.

Reply:
Works for me :+1: